jetty-core/jetty-http2/jetty-http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/internal/MetaDataBuilder.java

```diff
@@ -297,4 +303,20 @@ public MetaData build() throws HpackException.StreamException
             _contentLength = -1;
         }
     }
+
+    private HttpURI newHttpURI() throws HpackException.StreamException
+    {
+        try
+        {
+            return HttpURI.build()
+                .scheme(_scheme)
+                .host(_authority == null ? null : _authority.getHost())
+                .port(_authority == null ? -1 : _authority.getPort())
+                .pathQuery(_path);
```

Comment:
Do we need?
```suggestion
                .pathQuery(_path)
                .asImmutable();
```

Reply:
It creates another object in the hot path, and the return type is already just `HttpURI`, not its `Mutable` subclass.